### PR TITLE
fix(form-pill): fix issues where pill text not being truncated

### PR DESCRIPTION
**Pull request by krisantrobus**
Addresses an issue where form pill text would wrap when it is longer than form pill group parent.

Key considerations when working with FormPillGroup. As FormPillGroup wraps the pills there should be no ability to overflow the width of the formPillGroup, only to stretch as far as the parent group. In order to achieve this I need to find the string of the children as it could potentially be be composed like our storybooks. Apply the truncate only to the string. Currently for pill we would see the same issue with long text as we were already seeing in the original bug, it is not limited to being in a popover but to any narrow container.

Before:
Screenshot 2024-08-26 at 12 57 55 PM

After:
Screenshot 2024-08-26 at 1 13 51 PM
Screenshot 2024-08-26 at 1 22 55 PM

### CODE REVIEW

File: packages/paste-core/components/form-pill-group/package.json [hunk 1]
```diff
@@ -58,6 +59,7 @@
     "@twilio-paste/style-props": "^9.1.0",
     "@twilio-paste/styling-library": "^3.0.0",
     "@twilio-paste/theme": "^11.0.0",
```
Comment (PixeledCode) [inline packages/paste-core/components/form-pill-group/package.json:61]: any reason why `peerDependencies` has 14.0.0 version but `devDependencies` has 14.1.0 version of `@twilio-paste/truncate`?

Comment (krisantrobus) [reply packages/paste-core/components/form-pill-group/package.json:61]: This was the same versions for each in other components that used it to be consistent. Should resolve locally anyway

Comment (PixeledCode) [reply packages/paste-core/components/form-pill-group/package.json:61]: cool cool

File: packages/paste-core/components/combobox/stories/Combobox.stories.tsx [hunk 2]
```diff
@@ -885,3 +885,13 @@ ComboboxInModal.parameters = {
     disable: true,
   },
 };
+
+export const ComboboxErrorNarrowContainer: StoryFn = () => {
+  return (
+    <Box maxWidth="150px">
+      <Combobox items={items} labelText="Choose a component:" helpText="This is the help text" hasError />
```
Comment (serifluous) [inline packages/paste-core/components/combobox/stories/Combobox.stories.tsx:892]: suggestion: For VRT, can you show this one with the long option selected by default?

File: packages/paste-core/components/combobox/stories/MultiselectCombobox.stories.tsx [hunk 3]
```diff
@@ -688,3 +688,27 @@ export default {
...
+        }}
+        onSelectedItemsChange={(selectedItems) => {
+          // eslint-disable-next-line no-console
+          console.log(selectedItems);
+        }}
+      />
+    </Box>
+  );
+};
+MultiselectComboboxInNarrowContainer.storyName = "Narrow Contrainer";
```
Comment (serifluous) [inline packages/paste-core/components/combobox/stories/MultiselectCombobox.stories.tsx:714]:
```suggestion
MultiselectComboboxInNarrowContainer.storyName = "Narrow Container";
```

File: packages/paste-core/components/combobox/package.json [hunk 4]
```diff
@@ -55,6 +55,7 @@
     "@twilio-paste/styling-library": "^3.0.0",
     "@twilio-paste/text": "^10.0.0",
     "@twilio-paste/theme": "^11.0.0",
+    "@twilio-paste/truncate": "^14.0.0",
```
Comment (nkrantz) [inline packages/paste-core/components/combobox/package.json:58]: question: does combobox need truncate since you're just using textOverflow?

Comment (nkrantz) [reply packages/paste-core/components/combobox/package.json:58]: oh jk realizing it was added as a dependency for form pill group!

File: packages/paste-core/components/form-pill-group/stories/index.stories.tsx [hunk 5]
```diff
@@ -177,6 +179,157 @@ export const FormPillTreeVariant = (): JSX.Element => {
...
+              {pill}
+            </Text>
+          </FormPill>
+        ))}
+      </FormPillGroup>
+    </Box>
+  );
+};
+
+PillNarrowContainerFormattedTextContent.storyName = "Pill Narrow Container Formatted Text Content";
```
Comment (nkrantz) [inline packages/paste-core/components/form-pill-group/stories/index.stories.tsx:331]: suggestion(non-blocking) this line isn't necessary unless you're changing the name of the story, adds errors to the file

File: packages/paste-core/components/form-pill-group/stories/index.stories.tsx [hunk 6]
```diff
@@ -177,6 +179,157 @@ export const FormPillTreeVariant = (): JSX.Element => {
...
+          >
+            {index % 3 === 2 ? (
+              <Avatar
+                size={size === "large" ? "sizeIcon20" : "sizeIcon10"}
+                name="avatar example"
+                src="./avatars/avatar4.png"
+              />
+            ) : null}
+            {index % 3 === 1 ? <CalendarIcon decorative size={size === "large" ? "sizeIcon20" : "sizeIcon10"} /> : null}
+            <Text fontWeight="fontWeightBold" fontStyle="italic">
```
Comment (nkrantz) [inline packages/paste-core/components/form-pill-group/stories/index.stories.tsx:321]: issue: Text should have an `as` prop

File: packages/paste-core/components/form-pill-group/stories/index.stories.tsx [hunk 7]
```diff
@@ -177,6 +179,157 @@ export const FormPillTreeVariant = (): JSX.Element => {
...
+  React.PropsWithChildren<{
+    selected?: boolean;
+    dismissable?: boolean;
+    disabled?: boolean;
+    ariaLabel?: string;
+    size?: FormPillGroupSizeVariant;
+  }>
+> = ({ selected = false, dismissable = true, disabled = false, ariaLabel = "Basic pills:", size }) => {
+  const pillState = useFormPillState();
+  const pillState2 = useFormPillState();
```
Comment (nkrantz) [inline packages/paste-core/components/form-pill-group/stories/index.stories.tsx:298]: suggestion(non-blocking): should this line be removed? doesn't look like it's being used

File: packages/paste-core/components/form-pill-group/stories/index.stories.tsx [hunk 8]
```diff
@@ -177,6 +179,157 @@ export const FormPillTreeVariant = (): JSX.Element => {
...
+          </FormPill>
+        ))}
+      </FormPillGroup>
+    </Box>
+  );
+};
+
+PillNarrowContainerDecorationsEnd.storyName = "Pill Narrow Container Decorations End";
+
+export const PillNarrowContainerFormattedTextContent: React.FC<
```
Comment (nkrantz) [inline packages/paste-core/components/form-pill-group/stories/index.stories.tsx:288]:
```suggestion
export const PillNarrowContainerFormattedTextContent: StoryFn<
```

could you update the stories in this file to use StoryFn (imported from storybook/react)?

**Review by nkrantz — COMMENTED** [review]
I like this solution, still looks kind of weird if you try to format text inside of a form pill (with an ellipsis after each text element), but it's definitely an improvement from the current implementation. 
image

**Review by nkrantz — APPROVED** [review]

**Review by serifluous — APPROVED** [review]

**Pull request merged**